### DISCUSSION TITLE
Can't add test cases via trcli, test run is created but the xml file is not read so no test cases neither test results are visible in the Test Run

### DISCUSSION THREAD

**Discussion by emcence** [gurock/trcli#264] [Q&A] [answered]
Hello, 

I am following this guideline _https://support.testrail.com/hc/en-us/articles/12609674354068-Code-first-workflow#h_01H9GE9TJ6RD39FCR0FEW0FBBE_ to add test cases and results from playwright in TestRail. TestRun is created but TCs are not added and there is no test execution. In TestRail, **Automation ID** is configured under **Case Fields** with type **Text**

Command used 
`trcli -y -n -v \ -h "https://{{host}}.testrail.io" \ --username "" \ --password "" \ --project "TestProject" \ parse_junit --title "Playwright Automated Test Run" --case-matcher auto -f "/{{exact_location}}/results.xml"`

In the response can see:
**** Found 2 test cases not matching any TestRail case. Creating test run. **** 

Was expecting that TCs will be created in TestRail but complain is that they are not found

Have also tried to add properties in the xml file, but than it is not even able to parse the junit report
getting errors like:
```
File "{{exact_location}}/py_envs/bin/trcli", line 8, in <module>
sys.exit(cli())
^^^^^
File "{{exact_location}}/py_envs/lib/python3.12/site-packages/click/core.py", line 1157, in __call__
return self.main(*args, **kwargs)
^^^^^^^^^^^^^^^^^^^^^^^^^^
File "{{exact_location}}/py_envs/lib/python3.12/site-packages/trcli/cli.py", line 242, in main
return super().main(windows_expand_args=False, *args, **kwargs)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "{{exact_location}}/py_envs/lib/python3.12/site-packages/click/core.py", line 1078, in main
rv = self.invoke(ctx)
^^^^^^^^^^^^^^^^

```
Thanks if someone can help what need to be changed

**Comment by vshyshkin2024** [comment]
Have exact issue in attempt to publish test results

**Comment by jakec-routable** [comment, marked as answer]
In the command you posted, you have both `-y` and `-n` flags which are opposing flags for test case auto-creation. If you want your test cases to be auto-created, then only use `-y`, if not, then only use `-n`.

**Reply by emcence** [reply]
Works fine with -y. Thx

**Comment by vshyshkin2024** [comment]
Hi, Jake,

Is there limitation on Java version to be used in the project with

<dependency>
  <groupId>com.testrail</groupId>
  <artifactId>testrail-junit-extensions</artifactId>
  <version>0.2.0</version>
  <scope>test</scope>
</dependency>

Because for now one of the projects do use Java 21

And we saw very handy mechanism of adding attachments to the test result

Many thanks for support!

Viktor Shyshkin,
QA Lead,
Phoenix

**Reply by jakec-routable** [reply]
I couldn't say for certain. I haven't used that library yet. However, looking at the maven pom, it seems that the project is compiled with Java 1.8, so I think you would be ok to include it as a dependency in a Java 21 project because Java 21 comes well after Java 8 and should still be compatible. There is a small possibility that the library uses a feature that was removed in a later version of Java. In which case, you wouldn't be able to use it directly. You would discover this after trying to use the library as a dependency and encountering an error.

However, If you were planning on modifying and re-building the `testrail-junit-extensions`, then you would either need to do that with the Java 1.8 SDK since that's what the maven pom requires, or you can modify the pom and update the library to a Java 21 project and recompile it for your own project's usage.

**Comment by emcence** [comment]
Hi Jake, the problem with the upload of the TC is solved by using correct flag. But anyhow if in the test case additional properties are added like:

`      testInfo.annotations.push({ type: 'testrail_case_field', description: "First condition selected" });`

for each **test** and **test.step**

which give structure like this in the .xml file
```
<properties>
<property name="testrail_case_field" value="refs:TR-2">
</property>
<property name="testrail_case_field" value="First condition selected">
</property>
<property name="testrail_case_field" value="Correct interview is started">
</property>
</properties>
```

Which filed should be added in TestRail? Currently getting errors like:

```
File "{{exact_location}}/py_envs/bin/trcli", line 8, in <module>
sys.exit(cli())
^^^^^
File "{{exact_location}}/py_envs/lib/python3.12/site-packages/click/core.py", line 1157, in __call__
return self.main(*args, **kwargs)
^^^^^^^^^^^^^^^^^^^^^^^^^^
File "{{exact_location}}/py_envs/lib/python3.12/site-packages/trcli/cli.py", line 242, in main
return super().main(windows_expand_args=False, *args, **kwargs)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "{{exact_location}}/py_envs/lib/python3.12/site-packages/click/core.py", line 1078, in main
rv = self.invoke(ctx)
^^^^^^^^^^^^^^^^
```

Thanks in advance

**Reply by jakec-routable** [reply]
I haven't used the `testrail_case_field` property yet, but looking at the code, it seems that the `value` attribute of the `<property` should include the field name followed by a `:`, then the value, like
```
<property name="testrail_case_field" value="field_name:First condition selected" />
```
or
```
<property name="testrail_case_field">field_name:First condition selected</property>
```
If you still have trouble after that. You might try adding the `-v` flag to get more logging output, which may help identify the issue.

**Comment by vshyshkin2024** [comment]
Jake, Emilija,

Many thanks for your effort to help!!!
Our team will go further with TestRail and we are very glad that you
created such a product.
Will return to investigate the utilisation of that library next week as we
have a tight schedule for priority automation and for now simple
integration is ok.

Will back to you with some detailed issue if it will appear!
Again many thanks!

Viktor Shyshkin
QA Lead,
Phoenix365
